Fix support to MongoDB 4.x, 5.0 and 6.0

ecarruda opened the pull request:
Closes #34

marcuspoehls commented:
@ecarruda Hey Evandro, thank you for the merge request! I really appreciate your work. The tests are failing for MongoDB 4.x because the tests try to use `mongosh` instead of `mongo`. I can’t see the issue right now. Do you have an idea?

idorrington92 commented on start-mongodb.sh:
```diff
@@ -47,7 +52,7 @@ echo "::group::Waiting for MongoDB to accept connections"
 TIMER=0
 
-until docker exec --tty mongodb mongosh --port $MONGODB_PORT --eval "db.serverStatus()" # &> /dev/null
+until docker exec --tty mongodb $MONGO_CLIENT --port $MONGODB_PORT --eval "db.serverStatus()" # &> /dev/null
```
```suggestion
until docker exec --tty mongodb mongosh --port $MONGODB_PORT --eval "db.serverStatus()" # &> /dev/null
```
For mongoDB v4 this changes mongosh to mongo. Is that right? It might be why the tests are failing

marcuspoehls replied:
Hey Iain, MongoDB 4.x comes with a `mongo` command. Since MongoDB 5.x, there’s the newer `mongosh` command. The `$MONGO_CLIENT` variable should resolve the correct command depending on the defined MongoDB version.

ecarruda replied:
Let me check what happened with the Actions scripts, because locally everything worked, give me some time, probably some mess with shell environment vs my wildcard here:

```
if [[ $MONGODB_VERSION == 4* ]]; then
   MONGO_CLIENT="mongo"
 else
   MONGO_CLIENT="mongosh --quiet"
 fi
```

Working on it...

ecarruda commented:
Fixed wildcard problem sh vs bash, it worked locally on my MacOS even using and i didn't realize that I was using a bash wildcard without bash, sorry for the mess, fixed using shell script built in way. Waiting my local tests but I did some tests on linux and test workflow and all good. :)

marcuspoehls commented:
Awesome! Thank you for the refinements! 👍

ecarruda commented:
> Awesome! Thank you for the refinements! 👍

Thank you guys for the opportunity to help on this amazing actions solution to MongoDB in Github

ecarruda commented:
@marcuspoehls now all is good, just requested your review again and I also appreciate @idorrington92 review too :) Thank you guys. If you need some change or dislike the approach let me know.

idorrington92 commented:
That was a subtle bug! Thanks for fixing this so quickly @ecarruda

marcuspoehls commented on start-mongodb.sh:
```diff
@@ -18,6 +18,17 @@ if [ -z "$MONGODB_VERSION" ]; then
 
 
+echo "::group::Selecting correct MongoDB client"
+if [ "`echo $MONGODB_VERSION | cut -c1-1`" = "4" ]; then
```
@ecarruda Do you know if we can just use `1` instead of `1-1` for the first character?

```suggestion
if [ "`echo $MONGODB_VERSION | cut -c 1`" = "4" ]; then
```

Otherwise, I think a space between `-c` and `1-1` would be nice to increase readability. Well, hopefully my requested space doesn’t break anything.

ecarruda replied:
Let me do some tests here

ecarruda replied:
You are right @marcuspoehls thanks for this tip, checked on cut man page about this syntax and using -c alone (without --complement) we can improve this syntax as you suggested, changed it :)

marcuspoehls commented:
@ecarruda Just left a minor comment. Looks good overall. I’m happy to merge it. Will just wait in case you have an opinion related to my comment :)

ecarruda commented:
> @ecarruda Just left a minor comment. Looks good overall. I’m happy to merge it. Will just wait in case you have an opinion related to my comment :)

Yes, thanks. I'm always accept syntax improvements suggestion, I checked on cut man pages definitions and tested it here before and all good. Sent the change :) Thank you! Requested you review again

marcuspoehls commented:
@ecarruda Great 😃 I’ll merge this PR as soon as the test pipeline finishes

ecarruda commented:
> @ecarruda Great 😃 I’ll merge this PR as soon as the test pipeline finishes

Pretty good!!! Let me ask you, do you have any ideia/roadmap about when you will have another release with this change?

marcuspoehls commented:
New release comes in 10 minutes

marcuspoehls commented:
I’ve published version `1.8.0` to the GitHub Marketplace

ecarruda commented:
> I’ve published version `1.8.0` to the GitHub Marketplace

Thank you starting to update my workflows here to test it :)

marcuspoehls commented:
Cool! Please let me know how it works :)

ecarruda commented:
> I’ve published version `1.8.0` to the GitHub Marketplace

Just one thing strange, before i was using as your documentation but for some reason market is saying to use with `v` prefix now

```
- name: MongoDB in GitHub Actions
  uses: supercharge/mongodb-github-action@v1.8.0
```

ecarruda commented:
All working good here @marcuspoehls just this problem with the version tag, just let me know if you are going to do any change to rollback to `1.8.0` instead of `v1.8.0` then I decide my move here, if you are going to let with the `v` prefix some update on market documentation is also needed.

marcuspoehls commented:
@ecarruda Thank you for pointing this out! I’ve deleted the tag with the `v` prefix. And tagged a new version (including release) using just `1.8.0`.
